hack/tools/crd-generator/main.go

```diff
@@ -179,25 +180,51 @@ func runGenerator(args ...string) {
 	}
 }
 
-func opconTweaksMap(channel string, props map[string]apiextensionsv1.JSONSchemaProps) map[string]apiextensionsv1.JSONSchemaProps {
+// Apply Opcon specific tweaks to all properties in a map, and return a list of required fields according to opcon tags.
+// For opcon validation optional/required tags, the required list is updated accordingly.
+func opconTweaksMap(channel string, props map[string]apiextensionsv1.JSONSchemaProps, existingRequired []string) (map[string]apiextensionsv1.JSONSchemaProps, []string) {
```

Comment:
It would likely be more future proof to pass a pointer of the parent into this function and apply mutations directly. The current style in the PR would mean that any other tweaks that live at the parent level would mean more fields would have to be passed back and forth. And that _could_ get pretty cumbersome.

Reply:
Conceded, and I deliberately wasn't opening the gate too wide at this juncture, as its pretty simple to iterate to that if we _have to_, but my preference is to limit scope on what is an unexported, internal API.
However, that kinda represents saved state (as in "right now we know we limited things to avoid just passing a mutable parent... but we have to remember").  
That could get problematic in future as it is then incumbent on future reviewers to hold the line on parameter growth.  
Is there a way to make the desires more clear here?